qiskit/visualization/latex.py

```diff
@@ -73,88 +75,64 @@ def __init__(
         Raises:
             ImportError: If pylatexenc is not installed
         """
+        self._qubits = qubits
+        self._clbits = clbits
+
         # list of lists corresponding to layers of the circuit
-        self.nodes = nodes
+        self._nodes = nodes
 
         # image scaling
-        self.scale = 1.0 if scale is None else scale
+        self._scale = 1.0 if scale is None else scale
 
         # Map of cregs to sizes
-        self.cregs = {}
-
-        # List of qubits and cbits in order of appearance in code and image
-        # May also include ClassicalRegisters if cregbundle=True
-        self._ordered_bits = []
-
-        # Map from registers to the list they appear in the image
-        self.img_regs = {}
+        self._cregs = {}
 
         # Array to hold the \\LaTeX commands to generate a circuit image.
         self._latex = []
 
         # Variable to hold image depth (width)
-        self.img_depth = 0
+        self._img_depth = 0
 
         # Variable to hold image width (height)
-        self.img_width = 0
+        self._img_width = 0
 
         # Variable to hold total circuit depth
-        self.sum_column_widths = 0
+        self._sum_column_widths = 0
 
         # Variable to hold total circuit width
-        self.sum_wire_heights = 0
+        self._sum_wire_heights = 0
 
         # em points of separation between circuit columns
-        self.column_separation = 1
+        self._column_separation = 1
 
         # em points of separation between circuit wire
-        self.wire_separation = 0
+        self._wire_separation = 0
 
         # presence of "box" or "target" determines wire spacing
-        self.has_box = False
-        self.has_target = False
-        self.layout = layout
-        self.initial_state = initial_state
-        self.reverse_bits = reverse_bits
-        self.plot_barriers = plot_barriers
-
-        #################################
-        self._qubits = qubits
-        self._clbits = clbits
-        self._ordered_bits = qubits + clbits
-        self.cregs = {reg: reg.size for reg in cregs}
-
-        self._bit_locations = {
-            bit: {"register": register, "index": index}
-            for register in cregs + qregs
-            for index, bit in enumerate(register)
-        }
-        for index, bit in list(enumerate(qubits)) + list(enumerate(clbits)):
-            if bit not in self._bit_locations:
-                self._bit_locations[bit] = {"register": None, "index": index}
-
-        self.cregbundle = cregbundle
-        # If there is any custom instruction that uses clasiscal bits
+        self._has_box = False
+        self._has_target = False
+
+        self._reverse_bits = reverse_bits
+        self._plot_barriers = plot_barriers
+        self._layout = layout
+        self._initial_state = initial_state
+        self._cregbundle = cregbundle
+        self._global_phase = global_phase
+        self._circuit = circuit
```

Comment:
This is from `circuit_visualization.py`. I'm not sure what moving the setting of `layout` and `global_phase` into the drawers would buy you.
```
    if with_layout:
        layout = circuit._layout
    else:
        layout = None

    if fold is None:
        fold = 25

    global_phase = circuit.global_phase if hasattr(circuit, "global_phase") else None
    qcd = _matplotlib.MatplotlibDrawer(
        qubits,
        clbits,
        nodes,
        scale=scale,
        style=style,
        reverse_bits=reverse_bits,
        plot_barriers=plot_barriers,
        layout=layout,
        fold=fold,
        ax=ax,
        initial_state=initial_state,
        cregbundle=cregbundle,
        global_phase=global_phase,
        calibrations=circuit.calibrations,
        circuit=circuit,
    )
```

Reply:
It just buys you better data consistency; there's a single source of truth, and no opportunity for them to get out of sync if the code is expanded.  The data's already there in the class, so if we were re-writing the function, the question would be "why do we separate them out in the call?" - it's just that the development didn't quite go that way round.

Reply:
Ok. How about this. We do a new issue to push setting `qubits, clbits,, and layout` into the drawers along with appropriate deprecations, using `circuit` that's being passed into the drawers.

Reply:
Actually looks like `calibrations` can be pushed as well.

Reply:
So I think it makes sense to do everything except `qubits `and `clbits` here, since the deprecation process will all be similar. We'll put off `qubits` and `clbits ` until later since they might be a bit more complicated because of `idle_wires=False`. That's `cregs`, `qregs`, `global_phase`, `layout`, and `calibrations`.

Reply:
Ok. One more issue. If we deprecate `layout` as an arg to the drawer classes, we'd need to pass the `with_layout` bool to determine whether to use the `layout`. So which is preferred?
```
    if with_layout:
        layout = circuit._layout
    else:
        layout = None
```
from `circuit_visualization.py`.

Reply:
Your plan sounds good to me. I prefer passing `with_layout`, because it feels like a drawing option, whereas since `QuantumCircuit._layout` is a private field, it's better for users not to need to access it.